Add --syn option to iptables module and remove 'required: false' statements

siwyd opened the pull request:
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
iptables

##### ANSIBLE VERSION
N/A

##### SUMMARY
This adds the --syn option to filter SYN packets. Can be negated.

I added a generic append_match_flag function which can be used to add
match flags without parameters. It also allows negating the flag
if the added param allows this.

Not sure if I took the best approach here so all feedback welcome :)

Commit:
Add --syn option to iptables module

This adds the --syn option to filter SYN packets. Can be negated.

I added a generic append_match_flag function which can be used to add
match flags without parameters. It also allows negating the flag
if the added param allows this.

Not sure if I took the best approach here so all feedback welcome :)

LinusU commented:
Nice work!

'ship_it'

HyperBaton commented:
Cool, thanks for the quick merge!

dagwieers commented:
shipit

dagwieers commented on lib/ansible/modules/system/iptables.py:
```diff
@@ -226,6 +226,14 @@
+      - "This allows matching packets that have the SYN bit set and the ACK
+        and RST bits unset. When negated, this matches all packets with
+        the RST or the ACK bits set."
+    required: false
```
If it is not required, you don't have to add it. (The other parameters don't need `required: false` either)

dagwieers commented on lib/ansible/modules/system/iptables.py:
```diff
@@ -503,6 +529,10 @@ def main():
             reject_with=dict(required=False, default=None, type='str'),
             icmp_type=dict(required=False, default=None, type='str'),
+            syn=dict(
+                required=False,
```
Flag `required=False` is not needed, the is implied if it is not `required=True`

HyperBaton commented:
@dagwieers Okay, good to know. @LinusU Is it okay to remove all 'required: false' statements for all arguments?

LinusU commented:
Sounds very good 👍

Commit:
Remove unneccesary statements

* Remove 'required: false' statements from the argument docs
* Remove 'required=False' parameters from argument spec
* Remove 'default: null' statements from the argument docs

HyperBaton commented:
Whoops, embarrassing, I rebased wrongly, added 250 commits and now this pull request has a zillion tags ;) Apologies. I've forced pushed the corrections.

I've added a commit now to remove the unnecessary required statements. Stupid question perhaps, but is there an easy way to view/debug documentation changes like this?

simonwydooghe commented:
Could this merge request be reviewed by someone?

LinusU commented:
I think that required: false should have been removed from the dicts wheee the comment was placed, not from the documentation...

simonwydooghe commented:
@LinusU You're right, I only updated the documentation. I've now fixed the argument spec as well. I checked the documentation guidelines and it states that required: false is not needed and neither is default: null. But if you prefer to have it in there, let me know and I'll put it back in.

LinusU commented on lib/ansible/modules/system/iptables.py:
```diff
@@ -359,6 +327,13 @@ def append_param(rule, param, flag, is_list):
             rule.extend([flag, param])
 
 
+def append_match_flag(rule, param, flag, negatable):
```
Why have the `negatable` argument? It's always `True`?

siwyd replied:
So negatable means that the match param actually can be negated, meaning it's an option that works when you put a '!' in front of it. If a match option is not negatable, meaning it doesn't work with an exclamation mark it should get False as parameter value. So for syn it's set to True because it can be negated.

LinusU commented:
Looks very good, just one last question :)

simonwydooghe commented:
@LinusU Anything else needed here?

LinusU commented:
'ship_it'

siwyd committed:
Change version_added from 2.4 to 2.5

siwyd commented:
@Linusu I've had to rebase this as it seems other commits were merged in. Could you give this a review? I'd like to get this one merged. It's been in limbo for 6 months now :(

LinusU commented:
'ship_it'

sorry that it's taken so long :(

dagwieers committed:
Cosmetic changes to documentation and arg_spec

I didn't want to drag this any longer than it already has. Hope I am doing well.

dagwieers commented:
@LinusU The right command is **shipit**, but don't worry I will merge this if Shippable validate my cosmetic changes.

siwyd commented:
Thanks @dagwieers @LinusU!
